app/src/main/java/org/dasfoo/rover/android/client/util/L.java

```diff
@@ -0,0 +1,202 @@
+package org.dasfoo.rover.android.client.util;
+
+import android.util.Log;
+
+/**
+ * Log wrapper. To disable logging L.isLogEnable = false; To set min log level,
+ * when log enabled set L.logLevel = Log.VERBOSE;
+ */
+public class L {
+    /**
+     * It enables/disables logs.
+     */
+    public static boolean isLogEnable = false;
+
+    public static int logLevel = Log.VERBOSE;
+
+    /**
+     * Send a VERBOSE log message.
+     *
+     * @param tag    Used to identify the source of a log message
+     * @param string log message
+     */
+    public static void v(String tag, String string) {
+        if (isLogEnable && logLevel <= Log.VERBOSE) {
+            if (string == null) {
+                string = "null";
```

Comment:
I suggest having a method e.g.

```java
private static String stringOrNull(String s) {
  if (s == null) { return "null"; }
  return s;
}
```

and call `Log.v(tag, L.stringOrNull(string))` here.